zebra-network/src/protocol/external/types.rs

```diff
@@ -42,9 +42,43 @@ impl From<Network> for Magic {
 pub struct Version(pub u32);
 
 impl Version {
-    /// Returns the minimum network protocol version for `network` and
+    /// Returns the minimum remote node network protocol version for `network` and
+    /// `height`. Zebra disconnects from peers with lower versions.
+    ///
+    /// # Panics
+    ///
+    /// If we are incompatible with our own minimum remote protocol version.
+    pub fn min_remote_for_height(network: Network, height: block::Height) -> Version {
+        let min_spec = Version::min_specified_for_height(network, height);
+
+        // shut down if our own version is too old
+        assert!(
+            constants::CURRENT_NETWORK_PROTOCOL_VERSION >= min_spec,
+            "Zebra does not implement the minimum specified {:?} protocol version for {:?} at {:?}",
+            NetworkUpgrade::current(network, height),
+            network,
+            height,
+        );
+
+        max(min_spec, Version::initial_min_for_network(network))
+    }
+
+    /// Returns the minimum supported network protocol version for `network`.
+    /// This is the minimum peer version during initial block download.
+    fn initial_min_for_network(network: Network) -> Version {
+        Version::min_specified_for_upgrade(network, constants::INITIAL_MIN_NETWORK_PROTOCOL_VERSION)
+    }
+
+    /// Returns the minimum specified network protocol version for `network` and
+    /// `height`. This is the minimum peer version after initial block download.
+    fn min_specified_for_height(network: Network, height: block::Height) -> Version {
+        let network_upgrade = NetworkUpgrade::current(network, height);
+        Version::min_specified_for_upgrade(network, network_upgrade)
+    }
+
+    /// Returns the minimum specified network protocol version for `network` and
     /// `network_upgrade`.
-    pub fn min_for_upgrade(network: Network, network_upgrade: NetworkUpgrade) -> Self {
+    fn min_specified_for_upgrade(network: Network, network_upgrade: NetworkUpgrade) -> Self {
```

Comment:
Nit: maybe this should use `Version` instead of `Self` for consistency with the other methods? To be honest, my personal preference is for `Self` on method signatures, because it makes them shorter :laughing:, but I'm fine either way :+1:  

```suggestion
    fn min_specified_for_upgrade(network: Network, network_upgrade: NetworkUpgrade) -> Version {
```